tests/sagemaker/mock/__init__.py

```diff
@@ -514,13 +516,17 @@ def list_models(self):
             summaries.append(summary)
         return summaries
 
-    def list_tags(self, resource_arn, region_name):  # pylint: disable=unused-argument
+    def list_tags(
+        self, resource_arn, region_name, resource_type
+    ):  # pylint: disable=unused-argument
         """
         Modifies backend state during calls to the SageMaker "ListTags" API
         https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ListTags.html
         """
-        model = next(model for model in self.models.values() if model.arn == resource_arn)
-        return model.resource.tags
+        resource_values = getattr(self, resource_type).values()
+        for sagemaker_resource in resource_values:
```

Comment:
why is this being modified to a for loop? is ``self.resource_type.values()`` not accessible?

Reply:
It is not accessible because `resource_type` is not a valid attribute name. It's a variable containing a string value which is why line 526 was added. The original inline code was broken out to a for loop based on a previous review to improve readability.